src/Raven.Server/Documents/Handlers/AbstractStreamJsonFileBlittableQueryResultWriter.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Http;
+using Raven.Client;
+using Raven.Client.Util;
+using Raven.Server.Documents.Queries;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers;
+
+public abstract class AbstractStreamJsonFileBlittableQueryResultWriter<T> : IStreamQueryResultWriter<T>
+{
+    protected readonly AsyncBlittableJsonTextWriter Writer;
+    protected readonly string[] Properties;
+
+    protected AbstractStreamJsonFileBlittableQueryResultWriter([NotNull] HttpResponse response, [NotNull] Stream stream, [NotNull] JsonOperationContext context, string[] properties = null, string jsonFileNamePrefix = "export")
+    {
+        if (response == null)
+            throw new ArgumentNullException(nameof(response));
+        if (stream == null)
+            throw new ArgumentNullException(nameof(stream));
+        if (context == null)
+            throw new ArgumentNullException(nameof(context));
+
+        var encodedJsonFileName = Uri.EscapeDataString($"{jsonFileNamePrefix}_{SystemTime.UtcNow.ToString("yyyyMMdd_HHmm", CultureInfo.InvariantCulture)}.json");
+
+        response.Headers[Constants.Headers.ContentDisposition] = $"attachment; filename=\"{encodedJsonFileName}\"; filename*=UTF-8''{encodedJsonFileName}";
+        response.Headers[Constants.Headers.ContentType] = "text/json";
+
+        Properties = properties;
+        Writer = new AsyncBlittableJsonTextWriter(context, stream);
+    }
+
+    public ValueTask DisposeAsync()
+    {
+        GC.SuppressFinalize(this);
```

Comment:
I don't see any finalizer. Why do we have this call?

Reply:
This is a suggestion in VS to avoid calling finalizers, even if they are empty:
![image](https://github.com/ravendb/ravendb/assets/446555/9f12eee4-9be3-42bb-b426-29a5356151ae)